third_party/move/move-binary-format/src/file_format_common.rs

```diff
@@ -554,33 +557,25 @@ pub(crate) mod versioned_data {
                     .with_message("Bad binary header".to_string()));
             }
             let version = match read_u32(&mut cursor) {
-                Ok(v) => v,
+                Ok(v) => v & !APTOS_BYTECODE_VERSION_MASK,
                 Err(_) => {
                     return Err(PartialVMError::new(StatusCode::MALFORMED)
                         .with_message("Bad binary header".to_string()));
                 },
             };
             if version == 0 || version > u32::min(max_version, VERSION_MAX) {
-                return Err(PartialVMError::new(StatusCode::UNKNOWN_VERSION)
-                    .with_message(format!("bytecode version {} unsupported", version)));
-            } else if version == VERSION_NEXT
-                && !cfg!(any(test, feature = "testing", feature = "fuzzing"))
-            {
-                return Err(
-                    PartialVMError::new(StatusCode::UNKNOWN_VERSION).with_message(format!(
-                        "bytecode version {} only allowed in test code",
-                        VERSION_NEXT
-                    )),
-                );
+                Err(PartialVMError::new(StatusCode::UNKNOWN_VERSION)
+                    .with_message(format!("bytecode version {} unsupported", version)))
+            } else {
```

Comment:
I assume this is feature gated somewhere else, so It's ok to just return Ok

Reply:
It is, but still want to make stuff robust, also the feature gating is in Aptos code and not in move code.